nextstrain/cli/command/login.py

```diff
@@ -0,0 +1,153 @@
+        print("Checking credentials by logging into nextstrain.org...")
+        try:
+            config.read(secrets_path)
+            username = config.get('login', 'username')
+            authn_result = ast.literal_eval(config.get('login', 'authenticationresult'))
+
+        # If the requested credentials are not found in a 'login' section in the
+        # specified secrets file, generate them, and save them locally
+        except (FileNotFoundError, configparser.NoSectionError, configparser.NoOptionError):
+            login()
+            sys.exit(0)
+
+        u = Cognito(os.environ['AWS_COGNITO_USER_POOL_ID'],
+                '2vmc93kj4fiul8uv40uqge93m5',
+                username=username,
+                id_token=authn_result['IdToken'],
+                refresh_token=authn_result['RefreshToken'],
+                access_token=authn_result['AccessToken'])
+
+        try:
+            u.verify_token(authn_result['IdToken'], 'id_token', 'id')
+            u.verify_token(authn_result['AccessToken'], 'access_token', 'access')
+
+        except (jose.exceptions.JWSError, jose.exceptions.JWTError):
+            print("Invalid credentials.")
+            print()
+            login()
+            sys.exit(0)
+
+        access_token_expired = u.check_token()
+
+        if not access_token_expired:
+            print("Logged in succesfully!")
+
+        else:
+            print("Replacing expired access token...")
+            tokens['AccessToken'] = u.access_token
+
+            save_credentials({**{"username": username}, **tokens}, secrets_path)
+
+    else:
+        try:
+            config.read(secrets_path)
+            username = config.get('login', 'username')
+
+            print(f"Logged into nextstrain.org as {username}.")
+            print("Log out with `nextstrain logout`")
+
+        # If the requested credentials are not found in a 'login' section in the
+        # specified secrets file, generate them, and save them locally
+        except (FileNotFoundError, configparser.NoSectionError, configparser.NoOptionError):
+            login()
+
+
+def login():
+    """
+    With user input, logs a Nextstrain groups user into nextstrain.org.
+
+    Failed login attempts are printed to stdout and result in a non-zero exit
+    code.
+
+    Saves SRP credentials to a local *secrets_path*.
+    """
+    print("Logging into nextstrain.org...")
+    username = input('Username: ')
+    password = getpass.getpass()
+
+    try:
+        client = boto3.client('cognito-idp')
+
+        # Use AWSSRP utility class to easily save credentials
+        aws = AWSSRP(
+                username=username,
+                password=password,
+                pool_id=os.environ['AWS_COGNITO_USER_POOL_ID'],
+                client_id='2vmc93kj4fiul8uv40uqge93m5',
+                client=client
+            )
+        tokens = aws.authenticate_user()
+
+    except Exception as e:
+        print(f"Failed to login: {e}")
+        sys.exit(1)
```

Comment:
Better to throw a [`NextstrainUserError`](https://github.com/nextstrain/cli/blob/972803cd683ba7543d836f5d51a158ebe8034bff/nextstrain/cli/errors.py) instead of calling `sys.exit` directly so that deeper parts of the code aren't dead ends but pop the stack all the way back up with the exception. This allows a cleaner design with [a single point](https://github.com/nextstrain/cli/blob/972803cd683ba7543d836f5d51a158ebe8034bff/nextstrain/cli/__init__.py#L34-L39) to control expected exit statuses and error handling. (You may be familiar with the same concept people use in the React world, called "error boundaries" there.)